pkgs/development/python-modules/succulent/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, flask
+, pandas
+, pyyaml
+, poetry-core
+, pytestCheckHook
+, pythonOlder
+, toml-adapt
+}:
+
+buildPythonPackage rec {
+  pname = "succulent";
+  version = "0.2.3";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "firefly-cpp";
+    repo = "succulent";
+    rev = version;
+    hash = "sha256-MtUf3ZSti5kxVd/pJDcBtIXZvmthzCsMF7gSx/wJBcg=";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+    toml-adapt
+  ];
+
+  propagatedBuildInputs = [
+    flask
+    pandas
+    pyyaml
+  ];
+
+  # create flask and pandas versions consistent
+  preBuild = ''
+    toml-adapt -path pyproject.toml -a change -dep flask -ver X
+    toml-adapt -path pyproject.toml -a change -dep pandas -ver X
+  '';
```

Comment:
IMHO, we should stick to `pythonRelaxDepsHook` when dealing with issues related to dependencies. `toml-adapt` seems to be able to do the job but my guess is that most of the reviewers are not familiar with it.

Reply:
Thanks. Fixed now.